### PR TITLE
[FIX] Fix the binnding problem on scheduler during parallelization

### PR DESCRIPTION
**Pull request by xuyxu**
Issue: #33 

This PR fixes the problem that the deepcopy operation during parallelization will corrupt the bindings between scheduler and optimizer.

As an alternative solution, we can extract the `lr` from the scheduler and ingest it to the optimizer in the body of parallelizaiton.

### CODE REVIEW

**Review by xuyxu — COMMENTED** [review]
LGTM.

**Comment by xuyxu** [comment]
Below is the log after calling `model.set_scheduler("MultiStepLR", milestones=[2, 4])`. The initial learning rate is `1e-3`, and the float number printed below is `cur_lr` in the code.

The workflow on scheduler runs as expected now ;-)

```
0.001
2021-01-29 21:35:14,901 - INFO: Parallelization on the training epoch: 000
Estimator: 000 | Epoch: 000 | Batch: 000 | Loss: 2.30178 | Correct: 13/128
Estimator: 001 | Epoch: 000 | Batch: 000 | Loss: 2.29159 | Correct: 19/128
Estimator: 000 | Epoch: 000 | Batch: 100 | Loss: 1.89430 | Correct: 33/128
Estimator: 001 | Epoch: 000 | Batch: 100 | Loss: 1.82484 | Correct: 36/128
Estimator: 000 | Epoch: 000 | Batch: 200 | Loss: 1.83030 | Correct: 36/128
Estimator: 001 | Epoch: 000 | Batch: 200 | Loss: 1.72441 | Correct: 52/128
Estimator: 000 | Epoch: 000 | Batch: 300 | Loss: 1.71435 | Correct: 49/128
Estimator: 001 | Epoch: 000 | Batch: 300 | Loss: 1.71617 | Correct: 47/128
0.001
2021-01-29 21:35:31,443 - INFO: Parallelization on the training epoch: 001
Estimator: 000 | Epoch: 001 | Batch: 000 | Loss: 1.55694 | Correct: 52/128
Estimator: 001 | Epoch: 001 | Batch: 000 | Loss: 1.59709 | Correct: 49/128
Estimator: 000 | Epoch: 001 | Batch: 100 | Loss: 1.74537 | Correct: 46/128
Estimator: 001 | Epoch: 001 | Batch: 100 | Loss: 1.68254 | Correct: 53/128
Estimator: 000 | Epoch: 001 | Batch: 200 | Loss: 1.43353 | Correct: 64/128
Estimator: 001 | Epoch: 001 | Batch: 200 | Loss: 1.46144 | Correct: 62/128
Estimator: 000 | Epoch: 001 | Batch: 300 | Loss: 1.45001 | Correct: 60/128
Estimator: 001 | Epoch: 001 | Batch: 300 | Loss: 1.47741 | Correct: 62/128
0.0001
2021-01-29 21:35:44,893 - INFO: Parallelization on the training epoch: 002
Estimator: 000 | Epoch: 002 | Batch: 000 | Loss: 1.49852 | Correct: 58/128
Estimator: 001 | Epoch: 002 | Batch: 000 | Loss: 1.57914 | Correct: 48/128
Estimator: 000 | Epoch: 002 | Batch: 100 | Loss: 1.31425 | Correct: 72/128
Estimator: 001 | Epoch: 002 | Batch: 100 | Loss: 1.30448 | Correct: 65/128
Estimator: 000 | Epoch: 002 | Batch: 200 | Loss: 1.40527 | Correct: 60/128
Estimator: 001 | Epoch: 002 | Batch: 200 | Loss: 1.47308 | Correct: 61/128
Estimator: 000 | Epoch: 002 | Batch: 300 | Loss: 1.36096 | Correct: 64/128
Estimator: 001 | Epoch: 002 | Batch: 300 | Loss: 1.43065 | Correct: 55/128
0.0001
2021-01-29 21:35:58,446 - INFO: Parallelization on the training epoch: 003
Estimator: 000 | Epoch: 003 | Batch: 000 | Loss: 1.37930 | Correct: 67/128
Estimator: 001 | Epoch: 003 | Batch: 000 | Loss: 1.26449 | Correct: 67/128
Estimator: 000 | Epoch: 003 | Batch: 100 | Loss: 1.48528 | Correct: 59/128
Estimator: 001 | Epoch: 003 | Batch: 100 | Loss: 1.49262 | Correct: 58/128
Estimator: 000 | Epoch: 003 | Batch: 200 | Loss: 1.38797 | Correct: 60/128
Estimator: 001 | Epoch: 003 | Batch: 200 | Loss: 1.33060 | Correct: 64/128
Estimator: 000 | Epoch: 003 | Batch: 300 | Loss: 1.36791 | Correct: 70/128
Estimator: 001 | Epoch: 003 | Batch: 300 | Loss: 1.29499 | Correct: 71/128
1e-05
2021-01-29 21:36:12,452 - INFO: Parallelization on the training epoch: 004
Estimator: 000 | Epoch: 004 | Batch: 000 | Loss: 1.38219 | Correct: 64/128
Estimator: 001 | Epoch: 004 | Batch: 000 | Loss: 1.30229 | Correct: 78/128
Estimator: 000 | Epoch: 004 | Batch: 100 | Loss: 1.33359 | Correct: 68/128
Estimator: 001 | Epoch: 004 | Batch: 100 | Loss: 1.33889 | Correct: 69/128
Estimator: 000 | Epoch: 004 | Batch: 200 | Loss: 1.47949 | Correct: 54/128
Estimator: 001 | Epoch: 004 | Batch: 200 | Loss: 1.46495 | Correct: 60/128
Estimator: 000 | Epoch: 004 | Batch: 300 | Loss: 1.41752 | Correct: 57/128
Estimator: 001 | Epoch: 004 | Batch: 300 | Loss: 1.32704 | Correct: 72/128
1e-05
2021-01-29 21:36:26,097 - INFO: Parallelization on the training epoch: 005
Estimator: 000 | Epoch: 005 | Batch: 000 | Loss: 1.35294 | Correct: 60/128
Estimator: 001 | Epoch: 005 | Batch: 000 | Loss: 1.38788 | Correct: 58/128
```

**Review by zzzzwj — COMMENTED** [review]
LGTM
